pkg/provider/beyondtrust/provider.go

```diff
@@ -0,0 +1,360 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieclient.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package beyondtrust
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	auth "github.com/BeyondTrust/go-client-library-passwordsafe/api/authentication"
+	"github.com/BeyondTrust/go-client-library-passwordsafe/api/logging"
+	managed_account "github.com/BeyondTrust/go-client-library-passwordsafe/api/managed_account"
+	"github.com/BeyondTrust/go-client-library-passwordsafe/api/secrets"
+	"github.com/BeyondTrust/go-client-library-passwordsafe/api/utils"
+	"github.com/cenkalti/backoff/v4"
+	v1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	esoClient "github.com/external-secrets/external-secrets/pkg/utils"
+)
+
+const (
+	errNilStore             = "nil store found"
+	errMissingStoreSpec     = "store is missing spec"
+	errMissingProvider      = "storeSpec is missing provider"
+	errInvalidProvider      = "invalid provider spec. Missing field in store %s"
+	errInvalidHostURL       = "invalid host URL"
+	errNoSuchKeyFmt         = "no such key in secret: %q"
+	errInvalidRetrievalPath = "invalid retrieval path. Provide one path, separator and name"
+	errNotImplemented       = "not implemented"
+	errReturn               = "error: %w"
```

Comment:
There is zero purpose for this wrap. I would rather you properly wrap the errors or don't.